sycl/doc/extensions/experimental/sycl_ext_oneapi_fpga_mem.asciidoc

```diff
@@ -0,0 +1,506 @@
+
+=== Example
+[source,c++]
+----
+struct MyClass {
+  bool x;
+};
+
+using namespace sycl::ext::oneapi::experimental;
+
+sycl::queue q;
+q.single_task([=] {
+  fpga_mem<int[4], decltype(properties(resource<mlab>))> fm1 {1, 3, 5, 7};
+  fpga_mem<MyClass, decltype(properties(resource<mlab>))> fm2;
+  fm2.get().x = fm1[0];
+});
+----
+
+Above we see an example that declares `fpga_mem` variables `fm1`, `fm2`, and accesses them. 
+`fm1` creates an allocation on the device that is private to the kernel and contains an array of four `int` that is initialized to {1, 3, 5, 7}. `fm2` creates an allocation on the device that is private to the kernel and contains an object of type `MyClass` that is default-initialized. Both of variables will be implemented using the MLAB hardware resource. For more details on supported properties see a section below.
+
+`fpga_mem` has reference wrapper-like semantics, and is implicitly convertible to the wrapped type. Of note,
+because {cpp} doesn't allow for overloading of the "dot operator", a `get()`
+member of `fpga_mem` allows a reference to be extracted, to which the usual
+dot operator may be applied.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_FPGA_MEM` to one of the values defined in the table
+below. Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Representation of FPGA memory
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+template <typename T, typename PropListT = sycl::ext::oneapi::property_list<>>
+class fpga_mem {
+  ...
+----
+
+`fpga_mem` is a class template, parameterized by the type of the underlying
+allocation _T_, and a list of properties _PropListT_. The type of the
+allocation _T_ also encodes the size of the allocation.
+
+`fpga_mem` initializes the underlying object `T` with the values passed into it's constructor.
+
+Properties may be specified for a `fpga_mem` to provide semantic
+modification or optimization hint information to the compiler. See the section
+below for a list of the properties that are allowed.
+
+[NOTE]
+====
+
+`fpga_mem` has similar semantics to a reference wrapper.  The dot operator
+(`operator.`) cannot be overloaded, so a `get()` member is provided to allow a
+reference to be extracted directly when needed. Note that other operators can be overloaded by specific `T` as free functions,
+which will be selected through implicit conversion to `T` in device functions.
```

Comment:
I get what you're trying to say here but I think it's confusingly worded.  Specifically, it's not clear who has to write those overloads between: the end user, a library user, and the implementer of this extension.  I'd maybe say: "An fpga_mem templated on a type T can implicitly convert to T, thereby allowing it to be used in most places that an object of type T could be used.".